### PR TITLE
 add lightweight dev warning when theme is consumed but not provided

### PR DESCRIPTION
**Pull request by probablyup**
alternate fix for #2605

### CODE REVIEW

**Comment by haldunanil** [comment]
@probablyup Curious why you wanted to go with `console.warn` instead of throwing. Wouldn't it be easier, especially when you get a stacktrace on your screen, to see exactly what caused the error?

**Comment by probablyup** [comment]
Throwing is a bit aggressive IMO, could be swayed to make it a console error though. Also hopefully React will release 16.9 or a 16.8 patch soon that ships their warn and error utils that actually log the component stack for you :)

**Comment by haldunanil** [comment]
Ok that'll be dope! In that case `console.error` sounds better since it's not really a warning at that point as you're actually trying to reference something that the consumer component sees as being `undefined`, which will impede the way the component works as opposed to a warning which to me says "hey, this isn't great but carry on", which is not the case here-- you really should address asap.

**Comment by haldunanil** [comment]
Also would you mind if I updated my PR with your suggestions so I can have that coveted `Contributor` badge next to my name 😝?

**Comment by probablyup** [comment]
@haldunanil sure, this one is just for v5 anyway so you can do the corresponding version for v4

**Comment by haldunanil** [comment]
Cool, will open a new PR for that later today

File: packages/styled-components/src/models/StyledComponent.js [hunk 1]
```diff
@@ -261,6 +277,10 @@ export default function createStyledComponent(
   });
 
   if (process.env.NODE_ENV !== 'production') {
```
Comment (kitten) [inline packages/styled-components/src/models/StyledComponent.js]: So this is probably just a mildly controversial comment but I think we've previously been a bit heavy on warnings.

Yes, they may catch potential issues early, but at some point we'll have to think of the likelihood of them occurring.

Do we really think that this is going to provide value over a stack trace that is already clearly pointing to the same interpolation that could be causing this issue when a theme was missing?

**Comment by haldunanil** [comment]
@probablyup new PR for v4.0: https://github.com/styled-components/styled-components/pull/2657

File: packages/styled-components/src/models/StyledComponent.js (hunk 1 above)
Comment (probablyup) [reply packages/styled-components/src/models/StyledComponent.js]: There will only be a stack trace if they access a deeper property like `theme.colors.red`. We backfill `theme` with an empty object so `theme.color` wouldn't trigger a hard error, it'd just be undefined in the styles so the console error definitely would help catch bugs with more subtle usage.

File: packages/styled-components/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "styled-components",
-  "version": "5.0.0-beta.5",
+  "version": "5.0.0-beta.6",
```
Comment (kitten) [inline packages/styled-components/package.json]: Accidental bump in a PR?

Comment (probablyup) [reply packages/styled-components/package.json]: Just needs a rebase - it's a stray commit but one that's already in the canary branch 🤷‍♂

**Pull request merged**